include/picongpu/fields/background/templates/twtstight/TWTSTight.hpp

```diff
@@ -62,249 +62,219 @@
 #include <pmacc/math/Vector.hpp>
 #include <pmacc/types.hpp>
 
-namespace picongpu
+#include <array>
+#include <tuple>
+
+namespace picongpu::templates::twtstight
 {
-    namespace templates
+    using float_T = float_X;
+    using complex_T = alpaka::Complex<float_T>;
+    using complex_64 = alpaka::Complex<float_64>;
+    /** To avoid underflows in computation, numsigmas controls where a zero cutoff is made.
+     *  The fields thus are set to zero at a position (numSigmas * tauG * cspeed) ahead
+     *  and behind the respective TWTS pulse envelope.
+     *  Developer note: In case the float_T-type is set to float_64 instead of float_X,
+     *  numSigma can be increased to numSigmas = 10 without running into numerical issues.
+     */
+    constexpr uint32_t numSigmas = 6;
+
+    /** Provides the E- or B-field functors of the TWTSTight laser for the
+     *  fieldBackground and incidentField approaches.
+     *  @tparam T_Field Specializes for E- or B-Field
+     *  @tparam T_dim Specializes for the simulation dimension
+     */
+    template<typename T_Field>
+    struct TWTSTight
     {
-        namespace twtstight
-        {
-            using float_T = float_X;
-            /** To avoid underflows in computation, numsigmas controls where a zero cutoff is made.
-             *  The fields thus are set to zero at a position (numSigmas * tauG * cspeed) ahead
-             *  and behind the respective TWTS pulse envelope.
-             *  Developer note: In case the float_T-type is set to float_64 instead of float_X,
-             *  numSigma can be increased to numSigmas = 10 without running into numerical issues.
-             */
-            constexpr uint32_t numSigmas = 6;
+    public:
+        //! Helper method to define basic TWTS variables
+        HDINLINE static std::array<float_T, 11u> defineBasicHelperVariables(
```

Comment:
I think I would still have preferred a struct because we'd gain the advantage of having names for the variables, so it's slightly harder to mix them up. Decomposition into structured bindings would work exactly the same. But your solution achieves what we wanted to achieve, it's slightly shorter and the naming of variables doesn't really take effect because we're using structured bindings anyways, so no blocking objections.